api/db/migrate/20170425220220_add_counter_cache_to_collection.rb

```diff
@@ -0,0 +1,12 @@
+class AddCounterCacheToCollection < ActiveRecord::Migration[5.0]
+  def change
+    add_column :collections, :collection_resources_count, :integer, default: 0
+
+    execute <<-SQL.squish
+      UPDATE collections
+      SET collection_resources_count = (SELECT count(1)
```

Comment:
I know that I tell you to avoid using models in migrations, and that's still true. However, there is a rails method for what you're doing here, which is appropriate in this case: 

http://api.rubyonrails.org/classes/ActiveRecord/CounterCache/ClassMethods.html#method-i-reset_counters

`Resets one or more counter caches to their correct value using an SQL count query. This is useful when adding new counter caches, or if the counter has been corrupted or modified directly by SQL.`

We should use the more idiomatic rails method, IMO.